src/SubPool.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.26;
+
+import {Auth} from "./Auth.sol";
+import {ISubPoolFactory} from "./interfaces/ISubPoolFactory.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+import {ERC20} from "solady/tokens/ERC20.sol";
+
+contract SubPool is Auth {
+    error SubPool__OnlyFactory();
+    error SubPool__ExitDelayNotElapsedYet();
+    error SubPool__CollateralTokenAlreadyInitialized();
+    error SubPool__InsufficientETH();
+    error SubPool__CannotBillAfterExitDelay();
+    error SubPool__InvalidWithdraw();
+
+    /// @dev cannot be immutable because that allows for multiple pools per solver address.
+    address public collateralToken;
+    address public immutable COW;
+    ISubPoolFactory public immutable factory;
+
+    uint256 collateralDue;
+    uint256 cowDue;
+    uint256 ethDue;
+
+    constructor(address _owner, address _cow) {
+        factory = ISubPoolFactory(msg.sender);
+        COW = _cow;
+
+        // rely the owner
+        _addOwner(_owner);
+    }
+
+    function initializeCollateralToken(address token) external {
```

Comment:
The factory calls this method right after deploying this pool, so no one can hijack it, its all done in single tx.
Why should we have `minCowAmt` here? It is ensured during deployment that it is seeded with that amount and the user is not allowed to withdraw it until they exit.

Reply:
We shouldn't have `minCoWAmt` here (since as you said we have it in the factory). Instead we could also have `minCollateralAmount` here.

> The factory calls this method right after deploying this pool, so no one can hijack it, its all done in single tx.

Why is it written in two methods then? It requires the reader to also understand the factory. If the intention is to always call the two together, I think it would be more clear to have it in a single method.

Reply:
Why do we want the `minCollateralAmount` either? and who specifies it and to what? During pool deployment, user specifies how much to seed with and they cannot withdraw it either until exit.

Reply:
And it is a separate method because if you make the collateral part of the constructor the init code changes with collateral token and you cant enforce that there will only be 1 create2 address for the given user. You can have it checked with a mapping, but this is more gas efficient.

Reply:
> [...] you cant enforce that there will only be 1 create2 address for the given user.

Really good point! Can you add this as a comment in the code (so it's clear to future maintainers).

> Why do we want the minCollateralAmount either? 

I mean if it was for me I would remove all collateral amounts from the code, but for some reason we have minCowAmount in the factory (it looks like this is so that it can be read in `canSolve`). If this is the intention, then I think we should also specify and check the minimum collateral requirements for the other token. As to who is responsible to specify it, I'd say the same entity that's responsible to specify the minCowAmt. So you could have a mapping of "allowed collater -> minAmount" managed by the factory owner. But again, I personally think all of this is overkill and should be checked by whoever hands out the signatures off chain.

Reply:
We don't use the minCowAmount in `canSolve`, the `minCowAmount` is just there to enforce that atleast that much amount of COW is used to create new pools. We talked about it being a way to avoid spam, but I can remove the constraint just fine, just means the infrastructure would've to filter out pools periodically that only have tokens > $x or whatever

Reply:
> just means the infrastructure would've to filter out pools periodically that only have tokens > $x or whatever

I believe infra already has to do that. If a solver gets fined or doesn't have sufficient other collateral tokens it shouldn't send signatures either. So I think we should be consistent here and either enforce all amounts/tokens on the smart contract level (a lot of work) or get rid of all the requirements (I prefer). I'm not worried about sub pool spam frankly.

Reply:
removed `minCowAmt` and constraints